Validation notest

sylvestor88 opened the pull request.

Commit:
Capture Node Logs and zip older logs if any

Commit:
Merge branch 'validation-notest' of https://github.com/ptnapoleon/cassandra into validation-notest

ptnapoleon commented on test/validation/org/apache/cassandra/HarnessTest.java:
```diff
@@ -174,4 +177,14 @@ static URL getConfigURL(String yamlPath)
         }
     }
+
+    public String getTestName(String yamlPath){
```
Braces should always be on new lines.

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -17,9 +17,14 @@
+import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStreamReader;
+
```
This newline shouldn't be here.

ptnapoleon commented:
Make sure you follow the C\* style guide, especially with your imports:
http://wiki.apache.org/cassandra/HowToContribute

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -48,13 +56,19 @@ public CCMBridge(Config config)
+    }
+
+    public void destroy()
+    {
```
Why is stop() in captureLogs, and not in destroy?

sylvestor88 replied:
As I am capturing Node Logs before destroy(), so calling stop() from captureLogs() will include log data, if any generated by stop()

ptnapoleon replied:
While that is true, you're still breaking the functionality of destroy. Someone should feel comfortable calling destroy on the cluster without needing to call captureLogs or stop beforehand. It would be best to leave `stop` inside `destroy`. If you want to stop before capturing logs at the end of a test, just call the `stop` method, rather than embedding it inside captureLogs.

sylvestor88 replied:
Got it.. Made that change.

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -176,4 +190,102 @@ private String executeAndRead(String command, Object... args)
+        }
+    }
+
+    private static void compressFiles(ZipOutputStream zos,String folderName,String baseFolderName){
```
These last three functions are not specific to CCMBridge. They should be extracted out to a helper class.

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -27,9 +32,12 @@
 public class CCMBridge extends Bridge
 {
 
+    private int testNodeCount;
```
Rename this to nodeCount, please

Commit:
Node Logs Capture

Capture Node logs for a test and archive old logs

Capture Node Logs and zip older logs if any

Capture Node Logs and zip older logs if any

Capture Node Logs and zip older logs if any

Capture Node Logs and zip older logs if any

Capture Node Logs and zip older logs if any

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CaptureNodeLogs.java:
```diff
@@ -0,0 +1,126 @@
+
+public class CaptureNodeLogs
+{
+    public void preserveClusterLogs(File ccmDir, File CASSANDRA_DIR, String DEFAULT_CLUSTER_NAME, int nodeCount, String testName){
```
Move this back into CCMBridge. I said only the last three methods, not four, needed extracted.

ptnapoleon replied:
After that, just make all of this class's methods static.

sylvestor88 commented:
Pushed updated code

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CaptureNodeLogs.java:
```diff
@@ -0,0 +1,126 @@
+/*
+ *      Copyright (C) 2012-2014 DataStax Inc.
```
Totally wrong license header. Where did you copy this from?

sylvestor88 replied:
CCMBridge class

ptnapoleon replied:
Okay, that file has it's own license header because we copied that code from DataStax java driver. For literally every other class, use the license header found in literally ever other source code file. You just ran into bad luck here, nbd.

sylvestor88 replied:
Added license header same as HarnessTest

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -17,19 +17,24 @@
 import org.slf4j.LoggerFactory;
 import org.apache.cassandra.bridges.Bridge;
 import org.apache.cassandra.htest.Config;
+import org.apache.commons.io.FileUtils;
```
These imports aren't grouped correctly, but I just have my IDE handle that after I merge.

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -48,10 +53,18 @@ public CCMBridge(Config config)
 
+    public void captureLogs(String testName)
+    {
+        stop();
```
Again, I feel this is mixing concerns. You're violating Single Responsibility: https://en.wikipedia.org/wiki/SOLID_%28object-oriented_design%29 If you want the logs captured after a stop call, then you should be calling stop() prior to calling captureLogs() in the tearDown() of harnessTest.

sylvestor88 commented:
pushed changes.. Maybe I am no getting the stop() function right. However, I have added the function to abstract Bridge and calling it from the harness before clusterLogs(). I have removed stop() function from destroy() too.. Now when I think it, i guess I should have still kept stop() in the destroy function?

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -48,13 +51,18 @@ public CCMBridge(Config config)
+
     public void destroy()
     {
-        stop();
```
Please put this back.

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CaptureNodeLogs.java:
```diff
@@ -0,0 +1,98 @@
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class CaptureNodeLogs
```
These are all now just generic file manipulation methods, so the class should be renamed.

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CaptureNodeLogs.java:
```diff
@@ -0,0 +1,98 @@
+public class CaptureNodeLogs
+{
+
+    public static boolean checkForFolder(String dirPath){
```
Again, please keep all `{` on their own lines. It's part of the C\* style guide I linked in a previous diff. Set up your IDE to do so automatically.

sylvestor88 commented:
Pushed Updated Code

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -176,4 +185,35 @@ private String executeAndRead(String command, Object... args)
         }
     }
+
+    public void preserveClusterLogs(String testName)
```
What was the reasoning behind wrapping this method in `captureLogs`?

sylvestor88 replied:
Nothing more than for readability and re-usage purposes. Should I just define the whole thing in captureLogs() functions?

ptnapoleon replied:
I'm not opposed to extracting it, just curious about the reasoning. Though I do feel if it's going to be extracted like this, maybe the inner function shouldn't be public?

sylvestor88 commented:
I just implemented the whole thing in captureLogs() and took out the preserveClusterLogs()

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -176,4 +180,35 @@ private String executeAndRead(String command, Object... args)
+        String clusterLogsPath = ccmDir + "/" + DEFAULT_CLUSTER_NAME;
+        String folderName = testName;
+        String existingFolder = CASSANDRA_DIR + "/build/test/logs/validation/" + folderName;
+        ArchiveClusterLogs capture = new ArchiveClusterLogs();
```
All of this class's methods are static, there's no need to instantiate an object to do this.

sylvestor88 replied:
Yeah.. Makes sense. Made the changes

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java:
```diff
@@ -176,4 +180,35 @@ private String executeAndRead(String command, Object... args)
+            capture.zipExistingDirectory(existingFolder);
+        }
+
+        while (count <= nodeCount)
```
This is super nitpicky, but why not use a for loop here?

sylvestor88 replied:
No particular reason. Will for loop make any difference?

ptnapoleon replied:
Nope. You just happen to be instantiating a count variable prior to the loop, incrementing it at the end of the loop, and terminating the loop when you hit a certain count. It's perfectly suited for a for loop, and will look marginally cleaner.

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/ArchiveClusterLogs.java:
```diff
@@ -0,0 +1,97 @@
+                    FileInputStream in = new FileInputStream(folderName);
+                    int len;
+                    byte buffer[] = new byte[1024];
+                    while ((len = in.read(buffer)) > 0)
```
This is, perhaps, overly clever. It is not obvious to someone, unless they've read this part of the Java spec, that the assignment expression returns the new value of the assigned variable. Perhaps leave an in-line comment explaining this?

ptnapoleon commented on test/validation/org/apache/cassandra/bridges/ccmbridge/ArchiveClusterLogs.java:
```diff
@@ -0,0 +1,97 @@
+ * under the License.
+ *
+ */
+package org.apache.cassandra.bridges.ccmbridge;
```
Can you move which package this is in? Maybe out just to bridges? It's not specific to ccmbridge in any way.

sylvestor88 commented:
Pushed with new changes

ptnapoleon commented:
+1
